hs-init.hs

```diff
@@ -890,39 +886,82 @@ createStackTemplate
   -- create travis.yml template
   travisYml :: Text
   travisYml =
+    let travisMtr = T.concat (map travisMatrixItem testedVersions) in
     [text|
     {-# START_FILE .travis.yml #-}
-    # Use new container infrastructure to enable caching
-    sudo: false
+    sudo: true
+    language: haskell
+
+    git:
+      depth: 5
+
+    cache:
+      directories:
+      - "$$HOME/.stack"
+      - "$$HOME/build/${owner}/${repo}/.stack-work"
 
-    # Choose a lightweight base image; we provide our own build tools.
-    language: c
+    matrix:
+      include:
+
+      $travisMtr
+
+      - ghc: 8.2.2
+        env: GHCVER='8.2.2' STACK_YAML="$$HOME/build/${owner}/${repo}/stack.yaml"
 
-    # GHC depends on GMP. You can add other dependencies here as well.
     addons:
       apt:
+        sources:
+          - sourceline: 'ppa:hvr/ghc'
         packages:
-        - libgmp-dev
+          - libgmp-dev
 
     before_install:
-    # Download and unpack the stack executable
-    - mkdir -p ~/.local/bin
-    - export PATH=$$HOME/.local/bin:$$PATH
-    - travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+      - mkdir -p ~/.local/bin
+      - export PATH="$$HOME/.local/bin:$$PATH"
+      - travis_retry curl -L 'https://www.stackage.org/stack/linux-x86_64' | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+      - stack --version
 
-    # This line does all of the work: installs GHC if necessary, builds the
-    # library, executables, and test suites, and runs the test suites.
-    # `--no-terminal works` around some quirks in Travis's terminal implementation.
-    script: stack --no-terminal --install-ghc test
 
-    # Caching so the next build will be fast too.
-    cache:
-      directories:
-      - $$HOME/.stack
-      - $$HOME/build/$owner/${repo}/.stack-work
+    install:
+      - travis_wait 30 stack setup --no-terminal
+      - stack ghc -- --version
+
+      - travis_wait 40 stack build --only-dependencies --no-terminal
+      - travis_wait 40 stack build --test --bench --haddock --no-run-tests --no-run-benchmarks --no-haddock-deps --no-terminal
+
+    script:
+      - travis_wait 40 stack build --test --no-terminal
+
+    notifications:
+      email: false
+    $endLine
+    |]
+
+  travisMatrixItem :: Text -> Text
+  travisMatrixItem ghcV =
+    [text|
+    - ghc: ${ghcV}
+      env: GHCVER='${ghcV}' STACK_YAML="$$HOME/build/${owner}/${repo}/stack-$$GHCVER.yaml"
     $endLine
     |]
 
+  -- create specified @stack.yaml@ files
+  createStackYamls :: [Text] -> Text
+  createStackYamls = T.concat . map createStackYaml
+   where
+    createStackYaml :: Text -> Text
+    createStackYaml ghc =
+      let lts = case ghc of
+            "8.0.2"  -> "9.21"
+            "7.10.3" -> "6.35"
```

Comment:
Unfortunately, we build without `-Wall` so GHC don't warn you that pattern matching is not exhaustive here :(

I guess it's okay for now. But in some future we should have something like:

```haskell
data GhcVer = Ghc7103 | Ghc802 | Ghc822 ...

parseGhcVer :: Text -> Maybe GhcVer
```